api/v1alpha1/image_types.go

```diff
@@ -310,9 +310,7 @@ type ImageResourceSpec struct {
 type ImageFilter struct {
 	// name specifies the name of a Glance image
 	// +optional
-	// +kubebuilder:validation:MinLength:=1
-	// +kubebuilder:validation:MaxLength:=1000
-	Name *string `json:"name,omitempty"`
+	Name *OpenStackName `json:"name,omitempty"`
```

Comment:
Yeah, this is fine. Technically backwards incompatible, but not worth a version bump for v1alpha1, especially as any configuration which didn't meet it wouldn't work anyway.

Reply:
nit, and **not relevant for this very PR**: We just discovered that different OpenStack modules have different rules for names. I wonder if we want to keep having a common `OpenStackName` rather than e.g. `NetworkingName` and `ContainerName` going forward.